File: 모의고사.cpp
#include <string>
#include <vector>
using namespace std;

int person1[5] = {1,2,3,4,5};
int person2[8] = {2,1,2,3,2,4,2,5};
int person3[10] = {3,3,1,1,2,2,4,4,5,5};

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> result(3);
    for(int i=0; i<answers.size(); i++){
        if(answers[i]==person1[i%5]) result[0]++;
        if(answers[i]==person2[i%8]) result[1]++;
        if(answers[i]==person3[i%10]) result[2]++;
    }

    int max_score = max(max(result[0], result[1]), result[2]);
     
    for(int i=0; i<3; i++){
        if(result[i]==max_score)
            answer.push_back(i+1);
    }
    return answer;
}

File: Programmers/모의고사.py
def solution(answers):
    answer = []
    a1 = [1, 2, 3, 4, 5]
    a2 = [2, 1, 2, 3, 2, 4, 2, 5]
    a3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    s1, s2, s3 = 0, 0, 0
    
    for i in range(len(answers)):
        if a1[i%5] == answers[i] : s1+=1
        if a2[i%8] == answers[i] : s2+=1
        if a3[i%10] == answers[i] : s3+=1

    k = max(s1, s2, s3)
    
    if(k==s1) :
        answer.append(1)
    if(k==s2) : 
        answer.append(2)
    if(k==s3) :
        answer.append(3)
    return answer
